lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -962,6 +997,88 @@ static LogicalResult buildAllocOp(ComponentLoweringState &componentState,
                     IntegerAttr::get(rewriter.getI1Type(), llvm::APInt(1, 1)));
   componentState.registerMemoryInterface(allocOp.getResult(),
                                          calyx::MemoryInterface(memoryOp));
+
+  assert(memtype.getElementTypeBitWidth() <= 64 &&
+         "element bitwidth should not exceed 64");
+  bool isFloat = !memtype.getElementType().isInteger();
+
+  auto shape = allocOp.getType().getShape();
+  int totalSize =
+      std::reduce(shape.begin(), shape.end(), 1, std::multiplies<int>());
+  // The `totalSize <= 1` check is a hack to:
+  // https://github.com/llvm/circt/pull/2661 We should instead never pass
+  // multi-dimensional memories to Calyx, as discussed in:
+  // https://github.com/calyxir/calyx/issues/907
+  if (!(shape.size() <= 1 || totalSize <= 1)) {
```

Comment:
`totalSize` is for:
https://github.com/llvm/circt/blob/cf45d9410f94922d59fb994826c74931b949fe4b/test/Conversion/SCFToCalyx/convert_memory.mlir#L769

where its `shape.size()` is 4, but it can be collapsed to a `memref<1xi32>`

Reply:
Great, could you add that as an example in the comment above? It makes it much clearer.